cri.go

```diff
@@ -111,14 +112,36 @@ func initCRIService(ic *plugin.InitContext) (interface{}, error) {
 }
 
 // validateConfig validates the given configuration.
-func validateConfig(c *criconfig.Config) error {
-	// It is an error to provide both an UntrustedWorkloadRuntime & define an 'untrusted' runtime.
-	if _, ok := c.ContainerdConfig.Runtimes[criconfig.RuntimeUntrusted]; ok {
-		if c.ContainerdConfig.UntrustedWorkloadRuntime.Type != "" {
-			return errors.New("conflicting definitions: configuration includes untrusted_workload_runtime and runtimes['untrusted']")
+func validateConfig(ctx context.Context, c *criconfig.Config) error {
+	if c.ContainerdConfig.Runtimes == nil {
+		c.ContainerdConfig.Runtimes = make(map[string]criconfig.Runtime)
+	}
+
+	// Validation for deprecated untrusted_workload_runtime.
+	if c.ContainerdConfig.UntrustedWorkloadRuntime.Type != "" {
+		log.G(ctx).Warning("untrusted_workload_runtime is deprecated, please use `untrusted` runtime in `runtimes` instead")
```

Comment:
the statement please use `untrusted` runtime has been confusing to users, because we don't actually do anything in our code with the `untrusted` runtimes[] name?, where before with the untrusted workload we did.  I think it's probably better to just say it's no longer necessary or supported and that they can manually configure and select any runtime they desire via the `runtimes` feature.

or... if we desire to continue promoting the specification of an `untrusted` runtime in runtimes... let's just add a field in the config that points to the untrusted runtime..  Maybe:  `default_untrusted_runtime_name` but with the Runtimes match feature.. don't really see a point to specifying untrusted workloads anymore, or maybe I've missed something?

Reply:
The untrusted workload pod annotation still needs to be supported. Since it is an api, it will take longer to deprecate it.

I prefer not adding a new field `default_untrusted_runtime_name`, because we want to deprecate the config, adding new config option just makes deprecating them even harder.

Reply:
The "untrusted" runtime was added in 1.2 https://github.com/containerd/cri/pull/891.

Reply:
ok.. then let's just explain it better in the doc to reduce confusion.